_posts/2016-12-28-Day_Twentyone.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: post
+title: 8th Light apprenticeship - Day Twenty-one | Demystifying Elixir macros
+categories: 8thLight apprenticeship
+---
+
+A disclaimer before we begin: I am not advocating the use of macros. Quite the
+opposite in fact, as it will hopefully be clear throughout the article. Deciding 
+to use macros in your code should be given serious consideration and justified
+by significant benefits.
+
+I do not like magic tricks in programming. If something is mysteriously happening, I
+get this uncomfortable feeling that I am not really in control of my code: what
+if the magical thing stops working? How will I fix it if I have no clue on how it
+operates?
+
+I am not saying that I want to understand everything that goes on in a program,
+down to electrons flowing inside the processor, the same way a car driver does
+not need to know how the engine and transmission system work.
+This is called information hiding and
+it is a great thing. What I am not happy with is magic that happens at the same
+level of abstraction that I am in. For example, the Elixir Plug library has a
+Router module that allows you -amongst other things- to define what to do when a
```

Comment:
I think these dashes should be em-dash

http://www.thepunctuationguide.com/em-dash.html